tests/unit/middlewares/withAuth.test.ts

```diff
@@ -0,0 +1,261 @@
+    query: {},
+    cookies: {},
+    body: null,
+    env: {},
+  });
+
+  mockedExtendedRequest.headers = {
+    cookie: `connect-account-session=${sealedJWE}`,
+  };
+  mockedExtendedRequest.rawHeaders = [
+    "Cookie",
+    `connect-account-session=${sealedJWE}`,
+  ];
+
+  return { mockedExtendedRequest, mockedResponse };
+}
+
+const mockedHandler: Handler = async (
+  _mockedExtendedRequest,
+  _mockedResponse,
+) => {
+  return _mockedResponse.end();
+};
+
+describe("withAuth", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("should redirect to the login flow if no UserCookie is provided", async (done) => {
+    expect.assertions(3);
+
+    const access_token = generateHS256JWS();
+    const sealedJWS = await sealJWS(
+      access_token,
+      "thisIsAWrongSaltToMakeUserCookieUndefined",
+    );
+
+    const { mockedExtendedRequest, mockedResponse } = mockedReqAndRes(
+      sealedJWS,
+    );
+
+    const withAuthCallback = wrapMiddlewares(
+      [withSession, withAuth],
+      mockedHandler,
+    );
+    await withAuthCallback(mockedExtendedRequest, mockedResponse);
+
+    expect(mockedResponse.statusCode).toEqual(HttpStatus.TEMPORARY_REDIRECT);
+    expect(spiedOnDecryptVerifyAccessToken).not.toHaveBeenCalled();
+    expect(mockedVerifyJWTCall).toEqual(1);
+
+    done();
+  });
+
+  describe("Refresh tokens", () => {
+    test("should redirect to the login flow if a `TokenExpiredError` exception is thrown and no user is provided", async (done) => {
+      expect.assertions(7);
+
+      const access_token = generateHS256JWS({
+        ...defaultPayload,
+        exp: Date.now() - 3600,
+      });
+
+      const sealedJWS = await sealJWS(access_token);
+
+      const { mockedExtendedRequest, mockedResponse } = mockedReqAndRes(
+        sealedJWS,
+      );
+
+      spiedOnDecryptVerifyAccessToken.mockImplementationOnce(async () => {
+        class TokenExpiredError extends Error {
+          name = "TokenExpiredError";
+        }
+
+        throw new TokenExpiredError();
+      });
+
+      spiedOnGetDBUserFromSub.mockImplementation(async () => {
+        return null;
+      });
+
+      const withAuthCallback = wrapMiddlewares(
+        [withSession, withAuth],
+        mockedHandler,
+      );
+      await withAuthCallback(mockedExtendedRequest, mockedResponse);
+
+      expect(spiedOnDecryptVerifyAccessToken).toHaveBeenCalled();
+      expect(spiedOnGetDBUserFromSub).toHaveBeenCalled();
+      expect(spiedOnRefreshTokensFlow).not.toHaveBeenCalled();
+      expect(mockedVerifyJWTCall).toEqual(1);
+      expect(spiedOnGetAndPutUser).not.toHaveBeenCalled();
+      expect(mockedResponse.statusCode).toEqual(HttpStatus.TEMPORARY_REDIRECT);
+      expect(mockedResponse.getHeader("location")).toBe("/");
+
+      done();
+    });
+
+    test.only("should refresh the user tokens if a `TokenExpiredError` exception is thrown and a user is provided, and should not redirect", async (done) => {
```

Comment:
Is it the only test that needs to be run here ? :)

Reply:
What do you mean :)?

Reply:
I'm just wondering why you put `.only` on this test. It means that it will be the only one to be run on this test suit, right ?

Reply:
Dear god.. Nothing can escape your eyes, padawan.

Reply:
@Oluane LIER!!!

![image](https://user-images.githubusercontent.com/29123047/102190029-c29df080-3eb7-11eb-85b2-a333c12d2623.png)

🤣

Reply:
🤣

Reply:
Not a liar, I said I was working on it hehe. I make daily basis progress :stuck_out_tongue: